src/rand_tangent.jl

```diff
@@ -44,6 +44,6 @@ function rand_tangent(rng::AbstractRNG, x::T) where {T}
             Tangent{T}(; NamedTuple{field_names}(tangents)...)
         end
     else
-        return NO_FIELDS
+        return NoTangent()
     end
```

Comment:
We should not do this surface level change.
We actually get to remove the whole outer loop here.
we no longer need `if length(field_names) > 0`